diff --git a/drivers/gpu/drm/meson/meson_drv.c b/drivers/gpu/drm/meson/meson_drv.c
index 940f577a3b27..25d072327f88 100644
--- a/drivers/gpu/drm/meson/meson_drv.c
+++ b/drivers/gpu/drm/meson/meson_drv.c
@@ -232,7 +232,7 @@ static const struct meson_vpu_config meson8_vpu_config = {
 	/* The maximum frequency of HDMI PHY on Meson8 and Meson8m2 is ~3GHz */
 	.limits = {
-		.max_hdmi_phy_freq = 297000000,
+		.max_hdmi_phy_freq = 2970000000ULL,
 	},
 	.attrs = (const struct soc_device_attribute []) {
 		{ .soc_id = "Meson8 (S802)" },

diff --git a/drivers/gpu/drm/meson/meson_vclk.c b/drivers/gpu/drm/meson_vclk.c
index 105c3389a26e..976211013e2b 100644
--- a/drivers/gpu/drm/meson/meson_vclk.c
+++ b/drivers/gpu/drm/meson/meson_vclk.c
@@ -1054,12 +1054,12 @@ static int meson_vclk_set_parent(struct meson_drm *priv,
 	return 0;
 }

-static int meson_vclk_set_rate_exclusive(struct meson_drm *priv,
-					 enum vpu_bulk_clk_id clk_id,
-					 unsigned int rate_khz)
+static int meson_vclk_set_rate_exclusive(struct meson_drm *priv,
+					 enum vpu_bulk_clk_id clk_id,
+					 unsigned int rate_hz)
 {
 	struct meson_vclk *vclk = &priv->vclk[clk_id];
-	unsigned long rate_hz = rate_khz * 1000;
+	unsigned long rate = rate_hz;
 	int ret;

 	/*
@@ -1070,7 +1070,7 @@ static int meson_vclk_set_rate_exclusive(struct meson_drm *priv,
 	if (vclk->prepared)
 		return -EBUSY;

-	ret = clk_set_rate_exclusive(vclk->clk, rate_hz);
+	ret = clk_set_rate_exclusive(vclk->clk, rate);
 	if (ret < 0)
 		return ret;

